Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -176,6 +176,10 @@ @implementation FSTSyncEngine {
   /** The remote store for sending writes, watches, etc. to the backend. */
   RemoteStore *_remoteStore;
 
+  /** The callback to sync engine events like incoming snapshots, errors, and online state change.
+   */
```

Comment:
Optional: just reuse the comment that was originally on the delegate?
```
A callback to be notified when queries being listened to produce new view snapshots or errors.
```
(I'm mostly concerned that the part about queries is missing in the new wording)